### DISCUSSION TITLE
Mnemonics feedback? (And why isn't the Error Sheet `shift+e`?)

### DISCUSSION THREAD

**Discussion by reagle** [saulpw/visidata#1872] [Q&A]
Learning the shortcuts from scratch in a systematic way (by creating an Anki deck) is an interesting experience. Sometimes, I get snagged on something such as why isn't the Error Sheet `shift+e`? In any case, I thought I'd share the mnemonics that I use on the modifier keys, which also have images (such as snakes and Zorro) that can be used in remembering other sequences. Other suggestions?

# `g`

prefix/modifer for:

- **global**; enlarge; boost
- mnemonics: go; god; glow; 
- all  _selected_  rows (**G**od's selected)
- all  _visible_  columns (**g**low)

# `z`

modifier/prefix for:

- **zoomed**; smaller; precise; alternative 
- mnemonics: Zorro, python's hiss "zz"; sleep "zzz"; Zion Park
- current cell (**Z**orro captured and placed in jail  _cell_  )
- current column (for climbing  _column_  of rock at **Z**ion Park)

# `g`,`z`

prefix for:

- global **and** specific
- mnemonics: gauze; gizmo; gaze; glaze; 
- selected **cells**
- selected **rows** and visible **columns**

**Comment by saulpw** [comment]
> why isn't the Error Sheet shift+e?

This is a great question, and I've thought about it a few times.  The original thinking was that the errors sheet is more like a system sheet than a data sheet--like Ctrl+P for statuses or Ctrl+T for threads (and I've thought about changing this to Shift+T as well).  Nice catch.

Shift+Z for splitpane fits with your Zorro theme btw...it splits the screen in half (this was part of the reason I chose it).

For `g` and `z` a source code comment actually says `embig'g'en, 'z'mallify` which may or may not be helpful.

I like most of your mnemonics!  Zion's column is a bit of a stretch though. :)

**Comment by reagle** [comment]
Hi Saul, thanks for the response.

On 4/27/23 03:27, Saul Pwanson wrote:
> The original thinking was that the errors sheet is more like a system sheet than a data sheet

Oh, okay, that makes sense.

> Shift+Z for splitpane fits with your Zorro theme btw...

Got ya! (Though I wonder if younger people even know who Zorro is?)

> Zion's column is a bit of a stretch though. 

True, but is there something that better associates "z" with columns? (Zion was GPT's idea 😉 )

**Comment by frosencrantz** [comment]
To me there are a several other mnemonics, but not related to the modifiers. Like the type commands are on the number row.   There are some opposite direction commands related to the matched characters like (), {}, etc.  Also, many commands involving directions (up, left, etc) involve hjkl.

@reagle will you share your Anki deck?  I think that would help others.  I have heard the number of key bindings has caused people to say the learning curve is too high for VisiData.   Things have gotten better with the menus, and guides will help.

**Reply by reagle** [reply]
@frosencrantz, that is my intention. It's [available on github](https://github.com/reagle/visidata2-anki) for feedback -- and I can make the csv exports available as well. I'm still honing it (getting close) and waiting for clarity on some of the joins example before sharing it on the Anki repository.
